homeassistant/components/sms/__init__.py

```diff
@@ -75,7 +76,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     _LOGGER.debug("Connecting mode:%s", connection_mode)
     gateway = await create_sms_gateway(config, hass)
     if not gateway:
-        return False
+        raise PlatformNotReady(f"Cannot find device {device}")
```

Comment:
`PlatformNotReady` can only be raised from a platform. We can raise `ConfigEntryNotReady` here.
```suggestion
        raise ConfigEntryNotReady(f"Cannot find device {device}")
```